src/alchemlyb/preprocessing/subsampling.py

```diff
@@ -97,14 +98,16 @@ def statistical_inefficiency(df, series=None, lower=None, upper=None, step=None)
 
         # calculate statistical inefficiency of series
         statinef  = statisticalInefficiency(series)
-
-        # we round up
-        statinef = int(np.rint(statinef))
-
-        # subsample according to statistical inefficiency
-        series = series.iloc[::statinef]
-
-        df = df.loc[series.index]
+        
+        #use the subsampleCorrelatedData function to get the subsample index
+        indices = subsampleCorrelatedData(series, g=statinef)
+        picked_time_index = []
+        #pick the time index for the pandas dataframe based on the python index from subsample
+        for s_index, s_index_pair in enumerate(series.index):
```

Comment:
This looks clunky and potentially slow (although that might not matter much). Isn't there a direct way to do this without the `for` loop?

(I don't really know what the data look like to say more here.)

Reply:
Yea, let me check if there is a better way to rewrite this

Reply:
Yes, this will be very slow. You generally want to avoid looping through the rows in a `DataFrame` or `Series`. You can directly extract all values for an integer-based index with ``series.iloc[indices]`` instead.

Reply:
@dotsdl yes, the `iloc[indices]` works. Thx!